pkg/apis/extensions/types.go

```diff
@@ -268,6 +268,11 @@ const (
 
 	// Replace the old RCs by new one using rolling update i.e gradually scale down the old RCs and scale up the new one.
 	RollingUpdateDeploymentStrategyType DeploymentStrategyType = "RollingUpdate"
+
+	// The Deployment controller is not going to orchestrate ReplicaSets for
+	// Deployments that use this strategy. Useful for delegating orchestration of
+	// certain Deployments to external controllers.
+	NoopDeploymentStrategyType DeploymentStrategyType = "Noop"
```

Comment:
Hooks are one case. For example, I run a cluster operator that:

1. Executes a pre-hook (pod or job) to verify the running version of a dependency eg. apiserver
2. If succeeded, rollout a deployment eg. my custom scheduler

I can't run the pre-hook as part of the podspec of the scheduler because I am running multiple instances of it so I need a different kind of lifecycle hook at the deployment level.

Or I have a service that is using a cache and I want the cache flushed every time before I scale up new pods:
1. Scale down old replica set
2. flush the cache (again via a pod or job)
3. Scale up new replica set

or I want to build a pipeline that facilitates image promotion:
1. Deployment "qa" rolls out a new image, runs tests on it
2. If succeeded, run a post-hook that promotes the image to Deployment "prod"
3. Cleanup "qa" pods.

More advanced deployment scenarios can be made possible with custom controllers like A/B or Blue/Green deployments but for those we may need some additional changes either in the deployment api or some cli tooling that is currently missing from kubectl.